### PR TITLE
Space add idea frontend

### PR DESCRIPTION
**Pull request by kimberlym4488**
- [x] Create an idea page (ideaCreate.handlebars) added at /space/:space_id/ideas,
- [x] Endpoint functioning at /space/:space_id/idea/:idea_id,
- [x] Seeds file added to idea for #members and roles. Allow null: true

### CODE REVIEW

**Review by jmichaelbrown8 — APPROVED** [review]
Looks great! Pulled the branch down and tested that the homepage is protected by login and that you must request access to others' spaces.

File: controllers/homeRoutes.js [hunk 1]
```diff
@@ -64,16 +65,28 @@ router.get('/space/:space_id/access', withNoMembership, withAuth, async (req, re
     res.status(400).json(err);
     console.log(err);
   }
+});
+
+// Get idea create page
+router.get('/space/:space_id/ideas', withAuth, async (req, res) => {
```
Comment (ac524) [inline controllers/homeRoutes.js:71]: Add the `withApprovedMembership` authentication helper. See `/space/:space_id` route

File: controllers/homeRoutes.js [hunk 2]
```diff
@@ -64,16 +65,28 @@ router.get('/space/:space_id/access', withNoMembership, withAuth, async (req, re
...
+    console.log(err);
+  }
 });
 
-// Create idea page
-router.get('/space/:space_id/idea', withAuth);
 
 // View a specific idea
-router.get('/idea/:id', withAuth, async (req, res) => {
+router.get('/space/:space_id/idea/:idea_id', withAuth, async (req, res) => {
```
Comment (ac524) [inline controllers/homeRoutes.js:85]: Add the `withApprovedMembership` authentication helper. See `/space/:space_id` route

**Review by ac524 — CHANGES_REQUESTED** [review]

File: views/ideaCreate.handlebars [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        class='form-control'
+        id='idea-content'
+        name='idea-content'
+        placeholder='Have you ever wanted to...?'
+      ></textarea>
+    </div>
+
+    <div class='col-auto'>
+      <label for='member-validate'>I'd like members to join my idea</label>
+      <input type='checkbox' id='member-validate' onclick='showNumber()' />
```
[the remark this answers is not in the document]
Comment (kimberlym4488) [reply views/ideaCreate.handlebars]: All set :)

File: views/ideaCreate.handlebars [hunk 4]
```diff
@@ -0,0 +1,74 @@
+<h3 class='space-id' data-id='{{space.id}}'>{{space.name}}</h3>
+<br />
+
+<div class='card card-body' style="border:none">
```
[the remark this answers is not in the document]
Comment (kimberlym4488) [reply views/ideaCreate.handlebars]: Done :)

File: controllers/homeRoutes.js (hunk 1 above)
Comment (kimberlym4488) [reply controllers/homeRoutes.js:71]: Done

File: controllers/homeRoutes.js (hunk 2 above)
Comment (kimberlym4488) [reply controllers/homeRoutes.js]: Done :)

File: views/layouts/main.handlebars [hunk 5]
```diff
@@ -1,102 +1,174 @@
...
+      src='https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/js/bootstrap.bundle.min.js'
+    ></script>
+    <script
+      src='https://cdn.jsdelivr.net/npm/popper.js@1.16.1/dist/umd/popper.min.js'
+    ></script>
+
+    {{! jquery min script }}
+    <script
+      src='https://cdn.jsdelivr.net/npm/jquery@3.5.1/dist/jquery.slim.min.js'
+    ></script>
```
Comment (ac524) [inline views/layouts/main.handlebars:51]: See my comments in Slack. We should remove the popper (included in bs bundle, and swap the order of jQuery and bootsrap. That will allow the collapsible to work as expected with no script in the view.

File: views/ideaCreate.handlebars [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        id='form-create-idea'
+        class='btn btn-primary shadow-sm'
+      >Create</button>
+    </div>
+  </form>
+</div>
+
+{{!-- collapsibles not working without this add on in this handlebars page --}}
+<script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/js/bootstrap.bundle.min.js"></script>
+</head>
```
Comment (ac524) [inline views/ideaCreate.handlebars:73]: Remove bs script and rogue </head> tag. See my other comment that should help the collapse work without this.

**Review by ac524 — CHANGES_REQUESTED** [review]

File: views/layouts/main.handlebars (hunk 5 above)
Comment (kimberlym4488) [reply views/layouts/main.handlebars:48]: All set!

File: views/ideaCreate.handlebars (hunk 6 above)
Comment (kimberlym4488) [reply views/ideaCreate.handlebars]: This one too :)

**Review by ac524 — APPROVED** [review]
Nice Work!

One last callout (not a blocker), but the form fields don't expand full with for mobile and get squished to half the width of the page. You can do something like `col-12` for full width mobile. Then combine it with `col-md-6` or a larger screen size to drop it to smaller when needed.

**Pull request merged**